iceoryx_posh/include/iceoryx_posh/internal/runtime/ipc_interface_base.hpp

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
Could this maybe be done in a separate step? I guess this can be automated and might make it harder if there are some files which already have the SPDX license identifier

Reply:
@dkroenke requested this in [another pull request](https://github.com/eclipse-iceoryx/iceoryx/pull/548#discussion_r573242722). This is the reason I added it. What about  checking that with `ament_copyright` in the CI once there was a pull request that touched all files?

Reply:
Checking for SPDX license statement is possible with `ament_copyright` by creating a custom template package for it.

Reply:
@dkroenke the question was nor for checking, but if this should initially be done in an automated way and if it complicates things if it's already partially in some files

Reply:
@elBoberido I already adjusted 53 files in my PR according to the SPDX license statement.
And you can find the files easily which already have the statement (or reverse and add it with `sed -i`). Here a little fish one liner which lists all files which do not have the SPDX statement:
```sh
for i in (find iceoryx_posh -type f ); if test (cat $i | grep "// SPDX-License-Identifier: Apache-2.0" | wc -l) = 0; echo $i; end ; end
```

Reply:
@elfenpiff yes, but if this is done incrementally, we have to check each PR for the presence of the SPDX license statement until ... well, forever. If you volunteer to do this, I'll ping you in each PR ;)